tools/src/main/scala/scala/scalanative/build/NativeLib.scala

```diff
@@ -198,7 +203,7 @@ private[scalanative] object NativeLib {
             .stripSuffix(jarExt)
         NativeLib(
           src = path,
-          dest = workDir.resolve(s"$nativeCodePrefix-$name-$index")
+          dest = workDir.resolve(s"$nativeCodePrefix-$name-$confHash-$index")
```

Comment:
I'm not sure if it's enough to fix the underlying issue of changing native config. It would work handling changes in native glue code, but it would not force emitting new LLVM IR and especially the thread-safe `Op.Module` lowering (in singlethreaded mode it's a simple `get or else init` in multithreading mode is uses methods defined in `loadModule.c` to define a state machine waiting for the initialization of module by other thread.

Reply:
>  it would not force emitting new LLVM IR

This change actually makes scala-native emitting new LLVM IR 👍 

## Reproduction

- Apply `sandbox` to use `withMultithreadingSupport(false)`

```diff
diff --git a/project/Build.scala b/project/Build.scala
index 54890226..49564101 100644
--- a/project/Build.scala
+++ b/project/Build.scala
@@ -703,6 +703,11 @@ object Build {
       .enablePlugins(MyScalaNativePlugin)
       .withNativeCompilerPlugin
       .withJUnitPlugin
+      .settings(
+        nativeConfig ~= { c =>
+          c.withMultithreadingSupport(false)
+        },
+      )
       .dependsOn(scalalib, javalib, testInterface % "test")

 // Testing infrastructure ------------------------------------------------
```

- clean build, reload sbt settings, and run `nativeLink` on `sandbox3`

```sh
$ sbt
sbt> clean; reload
sbt> sandbox3 / nativeLink
```

- Checking if there's `__scalanative_loadModule` call / declare in LLVM IR

```sh

$ cd sandbox/.3/target/scala-3.3.1/native
$ grep -nr "__scalanative_loadModule" *.ll
# no matches
```

- Now, enable `multithreadingSupport` in `sandbox` 

```diff
diff --git a/project/Build.scala b/project/Build.scala
index 54890226..49564101 100644
--- a/project/Build.scala
+++ b/project/Build.scala
@@ -703,6 +703,11 @@ object Build {
       .enablePlugins(MyScalaNativePlugin)
       .withNativeCompilerPlugin
       .withJUnitPlugin
+      .settings(
+        nativeConfig ~= { c =>
+          c.withMultithreadingSupport(true)
+        },
+      )
       .dependsOn(scalalib, javalib, testInterface % "test")

 // Testing infrastructure ------------------------------------------------
```

- Reload (**do not clean!**), and `nativeLink` again.

```sh
sbt> reload
sbt> sandbox3 / nativeLink
```

- Checking if the LLVM IR are regenerated -> yes!

```
$ grep -nr "__scalanative_loadModule" *.ll | head -n 3
11b9624d.ll:76:declare i8* @"__scalanative_loadModule"(i8*, i8*, i64, i8*)
11b9624d.ll:2994:  %_5 = call i8* @"__scalanative_loadModule"(i8* %_4, i8* bitcast ({ { i8*, i8*, i32, i32, i8* }, i32, i32, { i8* }, [4 x i8*] }* @"_SM56scala.scalanative.runtime.ieee754tostring.ryu.RyuDouble$G4type" to i8*), i64 32, i8* bitcast (void (i8*)* @"_SM56scala.scalanative.runtime.ieee754tostring.ryu.RyuDouble$RE" to i8*)), !dbg !420
11b9624d.ll:5111:  %_5 = call i8* @"__scalanative_loadModule"(i8* %_4, i8* bitcast ({ { i8*, i8*, i32, i32, i8* }, i32, i32, { i8* }, [4 x i8*] }* @"_SM55scala.scalanative.runtime.ieee754tostring.ryu.RyuFloat$G4type" to i8*), i64 56, i8* bitcast (void (i8*)* @"_SM55scala.scalanative.runtime.ieee754tostring.ryu.RyuFloat$RE" to i8*)), !dbg !744
```